experimental/group/group.go

```diff
@@ -0,0 +1,50 @@
+package group
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/inngest/inngestgo/step"
+)
+
+type Result struct {
+	Error error
+	Value any
+}
+
+func Parallel(
+	ctx context.Context,
+	fns ...func(ctx context.Context,
+	) (any, error)) []Result {
+	ctx = context.WithValue(ctx, step.ParallelKey, true)
+
+	results := []Result{}
+	isPlanned := false
+	ch := make(chan struct{}, 1)
```

Comment:
Is this used as a waitgroup?

Reply:
It's effectively a waitgroup of 1. We want to run each step sequentially but also in a goroutine so that we can recover the control hijack

Reply:
and for cases where we want to execute each step in a different goroutine and join the results?